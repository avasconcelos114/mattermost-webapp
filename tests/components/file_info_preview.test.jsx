// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See License.txt for license information.

import React from 'react';

import {shallow} from 'enzyme';

import FileInfoPreview from 'components/file_info_preview.jsx';

describe('components/file_info_preview.jsx', () => {
    const requiredProps = {
        fileUrl: 'http://www.catphotos.org/wp-content/uploads/2013/02/cat-standing.jpg',
        fileInfo: {name: 'Test Image', size: 100, extension: 'jpg'}
    };

    test('should match snapshot', () => {
        const wrapper = shallow(
            <FileInfoPreview {...requiredProps}/>
        );

        expect(wrapper).toMatchSnapshot();
    });

    test('should receive props', () => {
        const wrapper = shallow(
            <FileInfoPreview {...requiredProps}/>
        );

        const instance = wrapper.instance();

        expect(instance.props.fileUrl).toBe(requiredProps.fileUrl);
        expect(wrapper.find('.file-details__name').text()).toBe(requiredProps.fileInfo.name);
    });
});