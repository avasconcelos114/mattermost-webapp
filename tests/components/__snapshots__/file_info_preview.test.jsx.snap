// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/file_info_preview.jsx should match snapshot 1`] = `
<div
  className="file-details__container"
>
  <a
    className="file-details__preview"
    href="http://www.catphotos.org/wp-content/uploads/2013/02/cat-standing.jpg"
    rel="noopener noreferrer"
    target="_blank"
  >
    <span
      className="file-details__preview-helper"
    />
    <img
      src={null}
    />
  </a>
  <div
    className="file-details"
  >
    <div
      className="file-details__name"
    >
      Test Image
    </div>
    <div
      className="file-details__info"
    >
      File type JPG, Size 100B
    </div>
  </div>
</div>
`;
